### PR TITLE
[dev docs] clarify local network instructions

### PR DESCRIPTION
**Pull request by sblackshear**
- More explicit instructions for restoring an old ledger state
- Removed some redundant wording/explanations
- Got rid of a confusing section at the end + moved the valuable content elsewhere

### CODE REVIEW

File: developers.diem.com/docs/core/run-local-network.md [hunk 1]
```diff
@@ -65,30 +64,17 @@ validator_1  | 	FullNode network: /ip4/0.0.0.0/tcp/7180
...
 
-In another terminal, start a `diem-client` using data from the output above:
+Use the output from above to start a `diem-client` in another terminal:
 
 ```
 $ cd ~/diem
 $ cargo run -p cli -- -c $CHAIN_ID -m $ROOT_KEY -u http://127.0.0.1:8080 --waypoint $WAYPOINT
 ```
 
-At the end of this tutorial you will have:
```
Comment (sblackshear) [inline developers.diem.com/docs/core/run-local-network.md:82]: This looks like a leftover from the beginning of an old doc--got rid of it.

File: developers.diem.com/docs/core/run-local-network.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ title: Run a Local Network
 
 You can run a local test validator network to test and debug services you are developing for the Diem Blockchain and to build and test Move module code changes. This local test network is not part of the Diem Ecosystem and is only meant for testing and development purposes.
```
Comment (gregnazario) [inline developers.diem.com/docs/core/run-local-network.md:6]: While we're here, let's fix this run on sentence.

`You can run a local test validator network to test and develop against a Diem Blockchain, including Move changes.  This network is not part of the Diem Ecosystem and is only for testing and development purposes.

File: developers.diem.com/docs/core/run-local-network.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ title: Run a Local Network
...
 
-You can use the Diem CLI command dev to compile, publish, and execute Move Intermediate Representation (IR) programs on your local test validator network.
+You can also use the Diem CLI command dev to compile, publish, and execute Move programs on your local test validator network.
 
 >
 >Note: Your local test network of validator nodes will not be connected to the testnet, and will not be using the actual validator nodes on the Diem Blockchain.
```
Comment (gregnazario) [inline developers.diem.com/docs/core/run-local-network.md:11]: `Note: Your local test network will not be connected to testnet, or mainnet of the Diem Blockchain.`

**Review by gregnazario — APPROVED** [review]
Thanks for cleaning this up!

The English in this tutorial could use a proofread on the rest of it.  It's pretty verbose and strangely worded.

**Pull request merged**